tests/accelerators/test_accelerator_connector.py

```diff
@@ -415,3 +417,20 @@ def test_plugin_accelerator_choice(accelerator, plugin):
 
     trainer = Trainer(plugins=plugin, num_processes=2)
     assert isinstance(trainer.accelerator.training_type_plugin, DDPShardedPlugin)
+
+
+@pytest.mark.parametrize(
+    ["accelerator", "plugin"],
+    [('ddp', DDPPlugin), ('ddp_spawn', DDPSpawnPlugin), ('ddp_sharded', DDPShardedPlugin),
+     ('ddp_sharded_spawn', DDPSpawnShardedPlugin), ('deepspeed', DeepSpeedPlugin)],
+)
```

Comment:
Looks like all this needs is a skip condition if deepspeed is not available. 
You could do that with 

`pytest.param("deepspeed", DeepSpeedPlugin, marks=pytest.mark.xfail(reason="needs deepspeed"))`

I believe

Reply:
omg thank you! let me add the mocks back